src/Sulu/Bundle/ResourceBundle/Controller/FilterController.php

```diff
@@ -0,0 +1,214 @@
+    {
+        try {
+            $filter = $this->getManager()->save(
+                $request->request->all(),
+                $this->getLocale($request),
+                $this->getUser()->getId()
+            );
+            $view = $this->view($filter, 200);
+        } catch (FilterDependencyNotFoundException $exc) {
+            $exception = new EntityNotFoundException($exc->getEntityName(), $exc->getId());
+            $view = $this->view($exception->toArray(), 400);
+        } catch (MissingFilterException $exc) {
+            $exception = new MissingArgumentException(self::$entityName, $exc->getFilter());
+            $view = $this->view($exception->toArray(), 400);
+        } catch (MissingFilterAttributeException $exc) {
+            $exception = new MissingArgumentException(self::$entityName, $exc->getAttribute());
+            $view = $this->view($exception->toArray(), 400);
+        }
+
+        return $this->handleView($view);
+    }
+
+    /**
+     * Change a filter by the given id.
+     *
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param integer $id the attribute id
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function putAction(Request $request, $id)
+    {
+        try {
+            $filter = $this->getManager()->save(
+                $request->request->all(),
+                $this->getLocale($request),
+                $this->getUser()->getId(),
+                $id
+            );
+            $view = $this->view($filter, 200);
+        } catch (FilterDependencyNotFoundException $exc) {
+            $exception = new EntityNotFoundException($exc->getEntityName(), $exc->getId());
+            $view = $this->view($exception->toArray(), 400);
+        } catch (FilterNotFoundException $exc) {
+            $exception = new EntityNotFoundException($exc->getEntityName(), $exc->getId());
+            $view = $this->view($exception->toArray(), 404);
+        } catch (MissingFilterException $exc) {
+            $exception = new MissingArgumentException(self::$entityName, $exc->getFilter());
+            $view = $this->view($exception->toArray(), 400);
+        }
+
+        return $this->handleView($view);
+    }
+
+    /**
+     * Delete an filter with the given id.
+     *
+     * @param integer $id the attribute id
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function deleteAction($id)
+    {
+        try {
+            $this->getManager()->delete($id);
+            $view = $this->view($id, 204);
+        } catch (FilterNotFoundException $exc) {
+            $exception = new EntityNotFoundException($exc->getEntityName(), $exc->getId());
+            $view = $this->view($exception->toArray(), 404);
+        }
+
+        return $this->handleView($view);
+    }
+
+    /**
+     * returns all fields that can be used by list
+     *
+     * @param \Symfony\Component\HttpFoundation\Request $request
```

Comment:
i think you will need here somethink like `@Get("collection/fields")`, when you will use it in the ui